tests/ballerina-test/src/test/java/org/ballerinalang/test/connectors/sql/SQLActionsTest.java

```diff
@@ -49,62 +57,101 @@
     private CompileResult resultNegative;
     private CompileResult resultMirror;
     private static final String DB_NAME = "TEST_SQL_CONNECTOR";
+    private DBType dbType;
+    private MySQLContainer mysql;
+    private PostgreSQLContainer postgres;
+    private BValue[] connectionArgs = new BValue[3];
+
+    @Parameters({"dataClientTestDBType"})
+    public SQLActionsTest(@Optional("HSQLDB") DBType dataClientTestDBType) {
+        this.dbType = dataClientTestDBType;
+    }
 
     @BeforeClass
     public void setup() {
+        switch (dbType) {
+        case MYSQL:
+            mysql = new MySQLContainer();
+            mysql.start();
+            String jdbcURL = mysql.getJdbcUrl();
+            String username = mysql.getUsername();
+            String password = mysql.getPassword();
```

Comment:
Shall we declare these variables outside the switch and move assigning to connectionArgs also to the outside?